### PR TITLE
Password resets, logout, change in flows, and more.

### PR DESCRIPTION
**Pull request by zheileman**
This PR introduce the password resets, the logout, and all the confirmation pages
for these, and for saving a case depending if it is on sign-up or on sign-in.

Also the user menu in the header is now mostly functional (except the change password).

Some pending work for extracting to a service object the handling of the linking of the
user to the tribunal case, and for sending the email (at the moment the email is being
sent only on account creation, not on existing account).

Copy need to be reviewed everywhere. At the moment, just sensible defaults.

Work in this and some previous PR relate to:

* https://www.pivotaltracker.com/story/show/143051113
* https://www.pivotaltracker.com/story/show/143051055
* https://www.pivotaltracker.com/story/show/143051095
* https://www.pivotaltracker.com/story/show/143050707

### CODE REVIEW

File: app/views/users/logins/signin_save_confirmation.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div class="grid-row">
+  <div class="column-two-thirds">
+    <h1 class="heading-large"><%=t '.heading' %></h1>
+
+    <p><%=t('.expire_warning', expire_in_days: Rails.configuration.x.cases.expire_in_days) %></p>
```
Comment (digitalronin) [inline app/views/users/logins/signin_save_confirmation.html.erb]: I think we will need to change this. The current text doesn't make it clear that it's 14 days from when you *started* the case.
Please add a TODO comment here to come back to this and use a helper method to say "Your case will be deleted in #{N} days," where N counts down from the case creation date.

File: app/views/users/registrations/new.html.erb [hunk 2]
```diff
@@ -6,9 +6,9 @@
   </i>
   <strong class="font-small">You must create an account before we can save your details.</strong>
```
Comment (digitalronin) [inline app/views/users/registrations/new.html.erb]: Can we move this text into the translation file, please?

File: app/views/users/registrations/signup_save_confirmation.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<div class="grid-row">
+  <div class="column-two-thirds">
+    <h1 class="heading-large"><%=t '.heading' %></h1>
+
+    <p class="lede"><%=t '.lead_text_html', user_email: @user_email %></p>
+
+    <p><%=t('.expire_warning', expire_in_days: Rails.configuration.x.cases.expire_in_days) %></p>
```
Comment (digitalronin) [inline app/views/users/registrations/signup_save_confirmation.html.erb]: Another TODO comment here, about counting down days (or specifying the absolute date on which we'll discard your case)

File: config/locales/en.yml [hunk 4]
```diff
@@ -1167,10 +1178,26 @@ en:
...
         heading: Sign into your account
         forgot_password: "I've forgotten my password"
+      logged_out:
+        heading: You have signed out
+        lead_text: Use the link emailed to you to return to your saved case.
+        more_text_html: <a href="https://www.gov.uk/">Go to GOV.UK</a>
+      signin_save_confirmation:
+        heading: Your appeal has been saved
+        lead_text_html: A confirmation email has been sent to:<br><strong>%{user_email}</strong>
+        expire_warning: We will keep a draft of your appeal for %{expire_in_days} days. Any information will be automatically deleted for your security after this time.
```
Comment (digitalronin) [inline config/locales/en.yml]: This is the same block of values, repeated twice. Can we move this to a YAML alias, please?

Comment (zheileman) [inline config/locales/en.yml]: I prefer not to until we know for sure those 2 pages will have the very same copy. Which I doubt.

File: app/views/users/registrations/new.html.erb (hunk 2 above)
Comment (zheileman) [inline app/views/users/registrations/new.html.erb]: Yep, I missed this one.

**Pull request merged**